### PR TITLE
fix: Return system theme when application theme not found

### LINKED ISSUE #20169: [Bug]: Return default theme if theme is not available
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

If the application's theme is not available due to some corrupted data in DB, the application is no longer usable. Instead of blocking the user from using the app, use the default theme in those cases.

### Steps To Reproduce

1. Open the application and set the theme
2. Customize the theme
3. Publish the application
4. Delete the published mode theme from database
5. Open the application as anonymous user in view mode

You'll see no theme is returned in the response to the `/api/v1/themes/applications/{applicationId}/current` API.

### Public Sample App

_No response_

### Issue video log

_No response_

### Version

Cloud 1.9.4

### PR DESCRIPTION
**Pull request by nayan-rafiq**
## Description
This PR returns the default theme if application theme is not available due to data corruption in DB.

Fixes #20169 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Manual
- Jest

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ThemeServiceCEImpl.java [hunk 1]
```diff
@@ -94,7 +95,8 @@ public Mono<Theme> getApplicationTheme(String applicationId, ApplicationMode app
                         themeId = application.getPublishedModeThemeId();
                     }
                     if (StringUtils.hasLength(themeId)) {
-                        return repository.findById(themeId, READ_THEMES);
+                        return repository.findById(themeId, Optional.empty())
```
Comment (nidhi-nair) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ThemeServiceCEImpl.java:98]: Should we use the deprecated lethod instead? Sidhant intends to revert the deprecation to favour code readability.

**Review by nidhi-nair — CHANGES_REQUESTED** [review]
Tiny comment

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ThemeServiceCEImpl.java (hunk 1 above)
Comment (nayan-rafiq) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ThemeServiceCEImpl.java]: @nidhi-nair I've updated. Would you please have a look?

**Review by nidhi-nair — APPROVED** [review]
Lgtm

**Pull request merged**